### PR TITLE
Ensure the shape of input patches is compatible with model constraints

### PR DESCRIPTION
**Pull request by fepegar**
This change uses TorchIO's [`EnsureShapeMultiple`](https://torchio.readthedocs.io/transforms/preprocessing.html#torchio.transforms.EnsureShapeMultiple) to ensure that the inputs are a multiple of certain number which is specified in the model size constraints.

### CODE REVIEW

**Review by Shruthi42 — APPROVED** [review]

File: Tests/ML/pipelines/test_inference_smallimages.py [hunk 1]
```diff
@@ -59,15 +59,6 @@ def run_inference_on_unet(size: TupleInt3) -> None:
         image_util.check_array_range(p)
 
 
-def test_inference_on_too_small_image() -> None:
-    """
```
Comment (javier-alvarez) [inline Tests/ML/pipelines/test_inference_smallimages.py:63]: Why delete a test?

Comment (fepegar) [reply Tests/ML/pipelines/test_inference_smallimages.py:63]: Because images will never be too small after merging
- #681

Comment (javier-alvarez) [reply Tests/ML/pipelines/test_inference_smallimages.py:63]: The other PR does not have tests. Is it not possible to have a test to demonstrate that this is correct?

**Review by javier-alvarez — APPROVED** [review]

File: Tests/ML/pipelines/test_inference_smallimages.py (hunk 1 above)
Comment (fepegar) [reply Tests/ML/pipelines/test_inference_smallimages.py:63]: I believe all this is covered by the tests in https://github.com/microsoft/InnerEye-DeepLearning/blob/main/Tests/ML/models/architectures/test_crop_size_multiple.py

**Review by ant0nsc — APPROVED** [review]

**Pull request merged**